### ISSUE TITLE
Add integration for GoodJob

### ISSUE THREAD

**Issue by olivier-thatch** [Shopify/job-iteration#453]
[GoodJob](https://github.com/bensheldon/good_job) is an alternative ActiveJob backend similar to Sidekiq and Resque.

I wanted to add an integration for GoodJob, but ran into an annoying issue: GoodJob uses Postgres as its data store, and job-iteration's test harness uses a MySQL database. So in order to be able to test it, I'd need to either switch everything to Postgres, or set up multiple databases.

Because neither of these options were very enticing, I opened a PR in GoodJob's own repository ([here](https://github.com/bensheldon/good_job/pull/1217)), but the maintainer understandably would prefer the integration to live in job-iteration's codebase.

How would you recommend I approach this? Would you be okay with switching the test database from MySQL to Postgres?

**Comment by olivier-thatch** [comment]
Hello,

We ended up using the following integration in our codebase:
```ruby
module JobIteration
  module Integrations
    module GoodJob
      class << self
        extend T::Sig

        sig { returns(T.nilable(T::Boolean)) }
        attr_accessor :stopping

        sig { returns(T::Boolean) }
        def call
          @stopping ||= T.let(false, T.nilable(T::Boolean))
          @stopping
        end
      end
    end

    ActiveSupport::Notifications.subscribe("scheduler_shutdown_start.good_job") do
      JobIteration::Integrations::GoodJob.stopping = true
    end

    JobIteration.interruption_adapter = JobIteration::Integrations::GoodJob
  end
end
```

Basically, we simply listen to the ActiveSupport notification sent by GoodJob on shutdown ([here](https://github.com/bensheldon/good_job/blob/131e0cc8cfca183b736d5bdab1e348a8801cea21/lib/good_job/scheduler.rb#L90)).

I wanted to submit this as a PR, however writing automated tests for this turned out to be quite complex: not only does GoodJob require a PostgreSQL database as mentioned in my first message, it also requires an actuals Rails app (there are [multiple references](https://github.com/search?q=repo%3Abensheldon%2Fgood_job+Rails.application+path%3A%2F%5Elib%5C%2F%2F&type=code) to `Rails.application` in its codebase). I did make an attempt at adding a dummy Rails app to job-iteration's test suite, but in the end it proved to be too much work, especially compared to the work involved in writing the actual integration (it is quite trivial). Hopefully someone more familiar with job-iteration's codebase and test suite will be able to write automated tests.
